### PR TITLE
Make the lint run on local paths (rather than git ls-tree output) by default

### PR DESCRIPTION
**Pull request by jgraham**
This change is [Reviewable](https://reviewable.io/reviews/w3c/wpt-tools/101)

### CODE REVIEW

File: lint/gitignore.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import re
+import fnmatch
+import os
+
+end_space = re.compile(r"([^\\]\s)*$")
+
+def fnmatch_translate(pat, path_name=False):
+    parts = []
+    seq = False
+    i = 0;
```
Comment (gsnedders) [inline lint/gitignore.py]: Gratuitous semicolon making the build fail.

**Comment by gsnedders** [comment]
Is there any reason why we can't just call `git-check-ignore` instead of trying to reimplement the matcher?

**Comment by jgraham** [comment]
Because the use case is "git isn't installed". This is important to Mozilla since we have to support pure-hg workflows.

**Comment by gsnedders** [comment]
Ah, okay. I for some reason thought we had other dependencies on git, but seemingly not.

FWIW, I'd rather we didn't introduce new untested lines without good reason, so I'd rather we actually tested `gitignore.py` in its entirety; I can live without `lint.py` being totally tested while it generally isn't for now, though…

Aside from that, it looks good.

**Comment by gsnedders** [comment]
Uh, we already have all of this, do we not?

**Comment by gsnedders** [comment]
@jgraham ^^

**Comment by gsnedders** [comment]
This all landed 8777171a5568e9fa967ef7a807dc400d89a07637.

**Pull request closed without merging**